File: .claude/worktrees/architecture-fix/src/backend/kirinchat/schemas/resume.py
from typing import Optional, Dict, List
from pydantic import BaseModel, Field


class ResumeInfoResp(BaseModel):
    """简历简要信息"""
    id: str = Field(..., description="简历ID")
    filename: str = Field(..., description="文件名")
    file_size: int = Field(..., description="文件大小")
    content_type: str = Field(..., description="MIME类型")
    status: str = Field(..., description="状态")
    score: Optional[float] = Field(default=None, description="评分")
    create_time: Optional[str] = Field(default=None, description="创建时间")


class ResumeDetailResp(BaseModel):
    """简历详情"""
    id: str = Field(..., description="简历ID")
    filename: str = Field(..., description="文件名")
    file_size: int = Field(..., description="文件大小")
    content_type: str = Field(..., description="MIME类型")
    status: str = Field(..., description="状态")
    score: Optional[float] = Field(default=None, description="评分")
    raw_text: Optional[str] = Field(default="", description="解析文本")
    analysis_result: Optional[Dict] = Field(default=None, description="分析结果")
    error_message: Optional[str] = Field(default="", description="错误信息")
    create_time: Optional[str] = Field(default=None, description="创建时间")


class ResumeStatusResp(BaseModel):
    """简历分析状态"""
    id: str = Field(..., description="简历ID")
    status: str = Field(..., description="状态")
    score: Optional[float] = Field(default=None, description="评分")


class ResumeListResp(BaseModel):
    """简历列表"""
    resumes: List[ResumeInfoResp] = Field(default=[], description="简历列表")

File: .claude/worktrees/architecture-fix/src/frontend/src/apis/resume.ts
import { request } from '../utils/request'
import type { UnifiedResponse } from './interview'

// ==================== Types ====================

export interface ResumeInfo {
  id: string
  filename: string
  file_size: number
  content_type: string
  status: string
  score: number | null
  create_time: string | null
}

export interface ResumeDetail extends ResumeInfo {
  raw_text: string
  analysis_result: Record<string, any> | null
  error_message: string
}

export interface ResumeListData {
  resumes: ResumeInfo[]
}

export interface ResumeStatusData {
  id: string
  status: string
  score: number | null
}

// ==================== APIs ====================

export const uploadResumeAPI = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return request<UnifiedResponse<ResumeInfo>>({
    url: '/api/v1/resume/upload',
    method: 'POST',
    data: formData,
    timeout: 120000,
  })
}

export const getResumeListAPI = () => {
  return request<UnifiedResponse<ResumeListData>>({
    url: '/api/v1/resume/list',
    method: 'GET',
  })
}

export const getResumeDetailAPI = (id: string) => {
  return request<UnifiedResponse<ResumeDetail>>({
    url: `/api/v1/resume/${id}`,
    method: 'GET',
  })
}

export const deleteResumeAPI = (id: string) => {
  return request<UnifiedResponse<{ success: boolean }>>({
    url: `/api/v1/resume/${id}`,
    method: 'DELETE',
  })
}

export const getResumeStatusAPI = (id: string) => {
  return request<UnifiedResponse<ResumeStatusData>>({
    url: `/api/v1/resume/${id}/status`,
    method: 'GET',
  })
}

export const getResumePdfUrl = (id: string) => {
  return `/api/v1/resume/${id}/pdf`
}
